p2p/switch.go

```diff
@@ -247,6 +247,7 @@ func (sw *Switch) OnStop() {
 //
 // NOTE: Broadcast uses goroutines, so order of broadcast may not be preserved.
 func (sw *Switch) Broadcast(e Envelope) {
+	_, _ = e.marshalMessage()
```

Comment:
Is this a noop? If yes, what's the point of having it?

Reply:
Calling `marshalMessage()` initializes the private field `messageBytes` in the `Envelope`. Then calls to `marshalMessage()` into subsequent functions will return the `messageBytes` without remarshaling it.